### DISCUSSION TITLE
infiniteQuery does not fetch next page

### DISCUSSION THREAD

**Discussion by ftanzarella-eviivo** [TanStack/query#4839] [Q&A] [answered]
Like the title explains, I can't seem to get the next set of results/page on request using infiniteQuery. I also made a snack from the example code in the docs but still nothing.

It looks like the pageParam never gets incremented when fetchNextPage is called. I've tried version 3 and 4 with same results. 

So I'm not sure if I should be incrementing the nextCursor manually. I can see nothing in the docs that suggests that I have to do so.

This is for React Native. The web examples on the docs work as expected.

[snack example](https://snack.expo.dev/@francot/frisky-edamame)

```const fetchProjects = async ({ pageParam = 0 }) => {
 const res = await fetch(`https://api.instantwebtools.net/v1/passenger? 
 page=${pageParam}&size=10`)
 return res.json()
}

const List = () => {
  const {
  data,
  error,
  fetchNextPage,
  hasNextPage,
  isFetching,
  isFetchingNextPage,
  status,
} = useInfiniteQuery({
  queryKey: ['test'],
  queryFn: fetchProjects,
  getNextPageParam: (lastPage, pages) => lastPage.nextCursor,
});

const loadMore = () => {
  fetchNextPage()
}

**Comment by TkDodo** [comment, marked as answer]
there is no property called `nextCursor` in the api response from this endpoint:

https://api.instantwebtools.net/v1/passenger?page=1&size=10

**Reply by ftanzarella-eviivo** [reply]
Okay I understand. "nextCursor" would be something the API returns that points to the next set of data (?)

Thanks
